lib/WebService/Async/Segment.pm

```diff
@@ -0,0 +1,229 @@
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::Segment - Unofficial support for the Segment service
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+This class acts as a B<Future>-based async Perl wrapper for segment HTTP API.
+
+=cut
+
+=head1 METHODS
+
+=head2 new
+
+Class constuctor of Segment wrapper class with following argunments:
+
+parameters:
+
+=over 4
+
+=item * C<write_key> - the API token of a Segment source.
+
+=item * C<base_uri> - the base uri of the Segment host, primarily useful for setting up test mock servers.
+
+=back
+
+=cut
+
+sub _init {
+    my ($self, $args) = @_;
+
+    for my $k (qw(write_key base_uri)) {
+        $self->{$k} = delete $args->{$k} if exists $args->{$k};
+    }
+
+    $self->next::method($args);
+}
+
+sub configure {
+    my ($self, %args) = @_;
+
+    for my $k (qw(write_key base_uri)) {
+        $self->{$k} = delete $args{$k} if exists $args{$k};
+    }
+
+    $self->next::method(%args);
+}
+
+=head2 write_key
+
+API token of the intended Segment source
+
+=cut
+
+sub write_key { shift->{write_key} }
+
+=head2 base_uri
+
+Server base uri as a C<URI> objects
+
+=cut
+
+sub base_uri {
+    my $self = shift;
+    return $self->{base_uri} if blessed($self->{base_uri});
+    $self->{base_uri} = URI->new($self->{base_uri} // SEGMENT_BASE_URL);
+    return $self->{base_uri};
+}
+
+=head2 ua
+
+A C<Net::Async::HTTP> object acting as HTTP user agent
+
+=cut
+
+sub ua {
+    my ($self) = @_;
+
+    return $self->{ua} if $self->{ua};
+
+    $self->{ua} = Net::Async::HTTP->new(
+        fail_on_error            => 1,
+        decode_content           => 1,
+        pipeline                 => 0,
+        stall_timeout            => 60,
+        max_connections_per_host => 2,
+        user_agent               => 'Mozilla/4.0 (WebService::Async::Segment; BINARY@cpan.org; https://metacpan.org/pod/WebService::Async::Segment)',
+    );
+
+    $self->add_child($self->{ua});
+
+    return $self->{ua};
+}
+
+=head2 basic_authentication
+
+Settings required for basic HTTP authentication
+
+=cut
+
+sub basic_authentication {
+    my $self = shift;
+
+    #C<Net::Async::Http> basic authentication information
+    return {
+        user => $self->write_key // '',
+        pass => ''
+    };
+}
+
+=head2 method_call
+
+Makes a Segment method call based on params. It automatically sets B<sentAt> to current time and B<context->{library}> to the current module.
+
+It takes two params:
+
+=over 4
+
+=item * C<method> - required. Segment method name (such as B<identify> and B<track>).
+
+=item * C<args> - optional. Method arguments represented as a hash. It may include either common, method-specific or custom fields.
+Please refer to L<https://segment.com/docs/spec/common/> for a full list of common fieds supported by Segment.
+
+=back
+
+=cut
+
+sub method_call {
+    my ($self, $method, %args) = @_;
+
+    $args{sentAt}                        = Date::Utility->new()->datetime_iso8601;
```

Comment:
```suggestion
    $args{sentAt}                        = delete($args{sent_at}) || Date::Utility->new->datetime_iso8601;
```

although I'd use Time::Moment for this module instead of Date::Utility, it's much faster.

Reply:
Changed to `Time::Moment->now_utc->to_string()`.